scale/source/models.py

```diff
@@ -33,6 +34,52 @@ def get_source_file_by_name(self, file_name):
 
         return ScaleFile.objects.get(file_name=file_name, file_type='SOURCE')
 
+    def get_source_jobs(self, source_file_id, started=None, ended=None, statuses=None, job_ids=None, job_type_ids=None,
+                        job_type_names=None, job_type_categories=None, error_categories=None, include_superseded=False,
+                        order=None):
+        """Returns a query for the list of jobs that have used the given source file as input. The returned query
+        includes the related job_type, job_type_rev, event, and error fields, except for the job_type.interface and
+        job_type_rev.interface fields.
+
+        :param source_file_id: The source file ID.
+        :type source_file_id: int
+        :param started: Query jobs updated after this amount of time.
+        :type started: :class:`datetime.datetime`
+        :param ended: Query jobs updated before this amount of time.
+        :type ended: :class:`datetime.datetime`
+        :param statuses: Query jobs with the a specific execution status.
+        :type statuses: [string]
+        :param job_ids: Query jobs associated with the identifier.
+        :type job_ids: [int]
+        :param job_type_ids: Query jobs of the type associated with the identifier.
+        :type job_type_ids: [int]
+        :param job_type_names: Query jobs of the type associated with the name.
+        :type job_type_names: [string]
+        :param job_type_categories: Query jobs of the type associated with the category.
+        :type job_type_categories: [string]
+        :param error_categories: Query jobs that failed due to errors associated with the category.
+        :type error_categories: [string]
+        :param include_superseded: Whether to include jobs that are superseded.
+        :type include_superseded: bool
+        :param order: A list of fields to control the sort order.
+        :type order: [string]
+        :returns: The list of jobs that match the time range.
+        :rtype: [:class:`job.models.Job`]
+        """
+
+        # Order must end with job ID so we can do a distinct on the job ID and remove duplicate jobs
+        if order:
```

Comment:
It might be good to add a unit test or two to cover these special case manipulations of the order parameter, although you do have one in the view tests, so not super urgent.